controllers/machinehealthcheck_targets.go

```diff
@@ -116,16 +117,22 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 
 	// the node has not been set yet
 	if t.Node == nil {
-		// status not updated yet
-		if t.Machine.Status.LastUpdated == nil {
-			return false, timeoutForMachineToHaveNode
+		// TODO change to checking ControlPlaneReadyCondition in v1alpha4, when cluster.spec.controlPlaneRef will be required.
```

Comment:
this should probably be ControlPlaneAvailable condition - when the first kubeadm API server is up and running -, assuming we are going to make this part of the contract (now AFAIK it exists only in KCP).

Reply:
Yeah, that makes more sense. I'll change it. And we should definitely make it a control plane provider contract requirement in v1alpha4.

Reply:
Has it been documented/decided anywhere that controlPlaneRef will be a requirement? It seems like that will create a barrier to phased adoption of Cluster API, though I suppose one could create the equivalent of a null controlplane provider

Reply:
Just to be clear, I'm not suggesting that we keep backward compat support for non-control plane provider managed control plane machines, just questioning the hard requirement on having a control plane provider implementation in place.

Reply:
@vincepri can you file an issue proposing controlPlaneRef as a requirement in v1alpha4